bundles/org.openhab.binding.wolfsmartset/src/main/java/org/openhab/binding/wolfsmartset/internal/handler/WolfSmartsetAccountBridgeHandler.java

```diff
@@ -0,0 +1,279 @@
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link WolfSmartsetAccountBridgeHandler} is responsible for managing
+ * communication with the WolfSmartset API.
+ *
+ * @author Bo Biene - Initial contribution
+ */
+@NonNullByDefault
+public class WolfSmartsetAccountBridgeHandler extends BaseBridgeHandler {
+    private static final int REFRESH_STARTUP_DELAY_SECONDS = 3;
+    private static final int REFRESH_INTERVAL_SECONDS = 1;
+    private static final int DEFAULT_REFRESH_INTERVAL_CONFIGURATION_MINUTES = 10;
+    private static final int DEFAULT_REFRESH_INTERVAL_VALUES_SECONDS = 15;
+
+    private final Logger logger = LoggerFactory.getLogger(WolfSmartsetAccountBridgeHandler.class);
+
+    private final HttpClient httpClient;
+
+    private @NonNullByDefault({}) WolfSmartsetApi api;
+    private @NonNullByDefault({}) String apiKey;
+    private int refreshIntervalConfigurationMinutes;
+    private int refreshIntervalValuesSeconds;
+    private boolean discoveryEnabled;
+
+    private final Map<String, WolfSmartsetSystemBridgeHandler> systemHandlers = new ConcurrentHashMap<>();
+    private final Set<String> systemIds = new CopyOnWriteArraySet<>();
+
+    private @Nullable Future<?> refreshSystemsJob;
+    private final AtomicInteger refreshConfigurationCounter = new AtomicInteger(REFRESH_STARTUP_DELAY_SECONDS);
+    private final AtomicInteger refreshValuesCounter = new AtomicInteger(REFRESH_STARTUP_DELAY_SECONDS);
+
+    public WolfSmartsetAccountBridgeHandler(final Bridge bridge, HttpClient httpClient) {
+        super(bridge);
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("AccountBridge: Initializing");
+
+        WolfSmartsetAccountConfiguration config = getConfigAs(WolfSmartsetAccountConfiguration.class);
+
+        Integer value;
+        value = config.refreshIntervalConfiguration;
+        refreshIntervalConfigurationMinutes = value == null ? DEFAULT_REFRESH_INTERVAL_CONFIGURATION_MINUTES : value;
+
+        value = config.refreshIntervalValues;
+        refreshIntervalValuesSeconds = value == null ? DEFAULT_REFRESH_INTERVAL_VALUES_SECONDS : value;
+
+        String username = config.username;
+        String password = config.password;
+        username = username == null ? "" : username;
+        password = password == null ? "" : password;
+
+        Boolean booleanValue = config.discoveryEnabled;
+        discoveryEnabled = booleanValue == null ? false : booleanValue.booleanValue();
+        logger.debug("AccountBridge: System and unit discovery is {}", discoveryEnabled ? "enabled" : "disabled");
+
+        try {
+            api = new WolfSmartsetApi(username, password, httpClient, scheduler);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Checking authorization");
+            scheduleRefreshJob();
+        } catch (WolfSmartsetCloudException e) {
```

Comment:
Please try to be more specific (login data wrong/missing connection/etc.) and do not force the user to interpret stacktraces, wherever possible. Stacktraces are usually better in deeper levels so a user with more knowledge can read them if needed. But a normal user is not able to read them.